katharsis-jpa/src/main/java/io/katharsis/jpa/query/criteria/JpaCriteriaExpressionFactory.java

```diff
@@ -0,0 +1,10 @@
+package io.katharsis.jpa.query.criteria;
+
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Expression;
+import javax.persistence.criteria.From;
+
+public interface JpaCriteriaExpressionFactory<T extends From<?, ?>> {
+
+	public Expression<?> getExpression(T parent, CriteriaQuery<?> query);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove usage of generic wildcard type. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1452)